src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -1010,77 +870,379 @@ private Assembly LoadAssemblyHelper(string assemblyName)
-                ErrorRecord errorRecord = new ErrorRecord(
-                    new Exception(String.Format(CultureInfo.CurrentCulture, AddTypeStrings.SpecialNetVersionRequired, Language.ToString(), string.Empty)),
-                    "LANGUAGE_NOT_SUPPORTED",
-                    ErrorCategory.InvalidArgument,
-                    Language);
+                _csharpCompilationOptions = new CSharpCompilationOptions(OutputAssemblyTypeToOutputKind(OutputType));
 
-                ThrowTerminatingError(errorRecord);
-                parseOptions = null;
+                if (UsingNamespace.Length > 0)
+                {
+                    _csharpCompilationOptions = _csharpCompilationOptions.WithUsings(UsingNamespace);
+                }
             }
 
-            SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(source, parseOptions);
-            var references = s_defaultAssemblies.Value;
-            if (ReferencedAssemblies.Length > 0)
+            if (String.Equals(ParameterSetName, "FromPath", StringComparison.OrdinalIgnoreCase) ||
+                String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase))
             {
-                var tempReferences = new List<PortableExecutableReference>(s_autoReferencedAssemblies.Value);
-                foreach (string assembly in ReferencedAssemblies)
+                foreach (string file in _paths)
                 {
-                    if (string.IsNullOrWhiteSpace(assembly)) { continue; }
-                    string resolvedAssemblyPath = ResolveAssemblyName(assembly, true);
+                    using (var sourceFile = new FileStream(file, FileMode.Open))
+                    {
+                        var sourceText = SourceText.From(sourceFile);
+                        _syntaxTrees.Add(CSharpSyntaxTree.ParseText(sourceText, _csharpParseOptions, file));
+                    }
+                }
+            }
+            else if (String.Equals(ParameterSetName, "FromMember", StringComparison.OrdinalIgnoreCase))
+            {
+                _sourceCode = GenerateTypeSource(Namespace, Name, _sourceCode, Language);
 
-                    // Ignore some specified reference assemblies
-                    string fileName = PathType.GetFileName(resolvedAssemblyPath);
-                    if (s_refAssemblyNamesToIgnore.Value.Contains(fileName))
+                var sourceText = SourceText.From(_sourceCode);
+                _syntaxTrees.Add(CSharpSyntaxTree.ParseText(sourceText, _csharpParseOptions));
+            }
+            else if (String.Equals(ParameterSetName, "FromSource", StringComparison.OrdinalIgnoreCase))
+            {
+                var sourceText = SourceText.From(_sourceCode);
+                _syntaxTrees.Add(CSharpSyntaxTree.ParseText(sourceText, _csharpParseOptions));
+            }
+            else
+            {
+                Diagnostics.Assert(false, "Invalid parameter set: {0}", this.ParameterSetName);
+            }
```

Comment:
nit: By looking at this assertion here, I feel maybe arrange this `if {} else if {} else if {} else {}` in a switch of parameter name would be cleaner.

Reply:
Done.